tests/compile-fail/borrow_box.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(clippy)]
+#![allow(boxed_local)]
+#![allow(blacklisted_name)]
+
+pub fn test(foo: &Box<bool>) { //~ ERROR you seem to be trying to use `&Box<T>`
```

Comment:
In the new tests, the type of a binding (e.g. `let foo: &Box<bool>`) is not detected by the lint. It seems like the `TypePass` needs an implementation for `check_expr` in order to catch this; is my understanding correct?

This will also have the effect of adding this check to the other two lints in the same pass (BOX_VEC and LINKEDLIST); that seems fine, but I just want to make sure that is desired.

I am also going to add a test for associated constants/types on a trait.